basic/conditional-area.test.js

```diff
@@ -0,0 +1,35 @@
+const Shape = require('./conditional-area');
+
+
+describe('ConditionalArea', function(){
+  
+  it('should calculate circle area', function(){
+    const shape = new Shape('circle');  
+    expect(shape.area({radius: 10})).toEqual(314.1592653589793);
+  });
+
+  it('should calculate rectangle area', function(){
+    const shape = new Shape('rectangle');   
+    expect(shape.area({width: 10, height: 5})).toEqual(50);
+  });
+
+  it('should calculate triangle area', function(){
+   const shape = new Shape('triangle');   
+    expect(shape.area({base: 2, height: 10})).toEqual(10);
```

Comment:
Try adding additional keys in the object and empty object, expect an error on it.

Reply:
Resolved 💯